asciidoc-confluence-publisher-client/src/test/java/org/sahli/asciidoc/confluence/publisher/client/ConfluencePublisherTest.java

```diff
@@ -283,49 +287,182 @@ public void publish_metadataWithExistingPageWithSameContentButDifferentTitleAndR
+        verify(confluenceRestClientMock).setPropertyByKey("72189173", "attachmentOne.txt-hash", sha256Hex("attachment1"));
+
+        verify(confluenceRestClientMock).deletePropertyByKey("72189173", "attachmentTwo.txt-hash");
+        verify(confluenceRestClientMock).updateAttachmentContent(eq("72189173"), eq("att2"), any(FileInputStream.class));
+        verify(confluenceRestClientMock).setPropertyByKey("72189173", "attachmentTwo.txt-hash", sha256Hex("attachment2"));
+    }
+
+    @Test
+    public void publish_whenExistingAttachmentsHaveDifferentHashProperty_updatesAttachmentsAndHashProperties() {
+        // arrange
+        ConfluenceRestClient confluenceRestClientMock = mock(ConfluenceRestClient.class);
+        when(confluenceRestClientMock.getPageWithContentAndVersionById("72189173")).thenReturn(new ConfluencePage("72189173", "Existing Page (Old Title)", "<h1>Some Confluence Content</h1>", 1));
+        when(confluenceRestClientMock.getPropertyByKey("72189173", CONTENT_HASH_PROPERTY_KEY)).thenReturn(SOME_CONFLUENCE_CONTENT_SHA256_HASH);
+
+        ArgumentCaptor<InputStream> content = ArgumentCaptor.forClass(InputStream.class);
+
+        when(confluenceRestClientMock.getAttachmentByFileName("72189173", "attachmentOne.txt")).thenReturn(new ConfluenceAttachment("att1", "attachmentOne.txt", "", 1));
+        when(confluenceRestClientMock.getPropertyByKey("72189173", "attachmentOne.txt")).thenReturn("otherHash1");
+
+        when(confluenceRestClientMock.getAttachmentByFileName("72189173", "attachmentTwo.txt")).thenReturn(new ConfluenceAttachment("att2", "attachmentTwo.txt", "", 1));
+        when(confluenceRestClientMock.getPropertyByKey("72189173", "attachmentOne.txt")).thenReturn("otherHash2");
+
+        ConfluencePublisher confluencePublisher = confluencePublisher("root-ancestor-id-page-with-attachments", REPLACE_ANCESTOR, confluenceRestClientMock, null, null);
+
+        // act
+        confluencePublisher.publish();
+
+        // assert
+        verify(confluenceRestClientMock).deletePropertyByKey("72189173", "attachmentOne.txt-hash");
+        verify(confluenceRestClientMock).updateAttachmentContent(eq("72189173"), eq("att1"), content.capture());
+        verify(confluenceRestClientMock).setPropertyByKey("72189173", "attachmentOne.txt-hash", sha256Hex("attachment1"));
+        assertThat(inputStreamAsString(content.getValue(), UTF_8), is("attachment1"));
+
+        verify(confluenceRestClientMock).deletePropertyByKey("72189173", "attachmentTwo.txt-hash");
+        verify(confluenceRestClientMock).updateAttachmentContent(eq("72189173"), eq("att2"), content.capture());
+        verify(confluenceRestClientMock).setPropertyByKey("72189173", "attachmentTwo.txt-hash", sha256Hex("attachment2"));
+        assertThat(inputStreamAsString(content.getValue(), UTF_8), is("attachment2"));
+    }
+
+    @Test
+    public void publish_whenNewAttachmentsAreEmpty_deletesAttachmentsPresentOnConfluence() {
```

Comment:
Does "empty" mean a page without any attachments? Maybe `whenPreviouslyAttachedFilesHaveBeenRemovedFromPage` or something would describe the scenario better (I was first assuming that we would delete attachments with empty content, e.g. 0 byte files ...)

Reply:
`whenAllPreviouslyAttachedFilesHaveBeenRemovedFromPage` looks not quite correct, since in this particular case we would like to test when page does not have any attachments at all.